### DISCUSSION TITLE
target_line = block[0].strip().split() TabError: inconsistent use of tabs and spaces in indentation

### DISCUSSION THREAD

**Discussion by Tangbbmc** [RosettaCommons/rosetta#160] [Rosetta]
I want to build the mutated antibody structure using RosettaCM. When running setup_RosettaCM.py, the following error happened: 
File "/home/dell/Downloads/rosetta.MPI/main/tools/protein_tools/scripts/setup_RosettaCM.py", line 149
    target_line = block[0].strip().split()
TabError: inconsistent use of tabs and spaces in indentation

**Comment by roccomoretti** [comment]
The RosettaCM scripts were likely built with Python2. As such, they may not be 100% compatible with Python3.

You can try running them with `python2` if you have it installed. If you don't, a Python2 environment should be installable with Conda. (I'd recommend getting familiar with Conda, as most of the newer biomolecular modeling packages default to Conda installation.)

Worse comes to worse, you can try to run [Python's 2to3 converter](https://docs.python.org/3/library/2to3.html) on the scripts to convert the usage to Python3. They probably shouldn't be doing anything too strange, so the automated conversion will likely work.

**Reply by Tangbbmc** [reply]
@roccomoretti Thank you for your suggestion! The error mentioned above disappeared when running with python2. But a new error was reported, see below:
**python2 /home/dell/Downloads/rosetta.MPI/main/tools/protein_tools/scripts/setup_RosettaCM.py --fasta mutA_seq.fasta --alignment mutA_WT.fasta --alignment_format fasta --template WT.pdb --rosetta_bin /home/dell/Downloads/rosetta.MPI/main/source/bin
sh: 1: /home/dell/Downloads/rosetta.MPI/main/source/bin/partial_thread.default.linuxgccrelease: not found
Traceback (most recent call last):
  File "/home/dell/Downloads/rosetta.MPI/main/tools/protein_tools/scripts/setup_RosettaCM.py", line 647, in <module>
    for name in thread_fullnames: assert (os.path.exists(name)), "File %s doesn't exist"%(name)
AssertionError: File /home/dell/Videos/rosetta_P570/rosetta_cm/P570__thread.pdb doesn't exist**


Notably, the installed version of rosetta is MPI version. The bin folder contains partial_thread.mpi.linuxgccrelease.

**Reply by roccomoretti** [reply]
You should be able to change what the 'extras' build the runner script uses by using the `--build` option (e.g. `--build mpi`) -- There's similar other options for compiler, platform, mode, etc. See `python2 /home/dell/Downloads/rosetta.MPI/main/tools/protein_tools/scripts/setup_RosettaCM.py --help` for more info.
